src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -967,17 +987,18 @@ void Replayer<I>::create_mirror_snapshot(
 
 template <typename I>
 void Replayer<I>::handle_create_mirror_snapshot(
-    int r, cls::rbd::GroupSnapshot *snap, Context *on_finish) {
-  dout(10) << snap->id << ", r=" << r << dendl;
+    int r, cls::rbd::GroupSnapshot local_snap, Context *on_finish) {
+  std::string group_snap_id = local_snap.id;
+  dout(10) << group_snap_id << ", r=" << r << dendl;
 
-  if (r < 0) { // clean the group snapshot here
-    dout(10) << "cleaning snapshot as failed group_snap_set previously with : "
-             << snap->id << ", this will be attempted to sync later" << dendl;
-    r = librbd::cls_client::group_snap_remove(&m_local_io_ctx,
-        librbd::util::group_header_name(m_local_group_id), snap->id);
-    if (r < 0) {
-      derr << "failed to remove group snapshot : "
-           << snap->id << " : " << cpp_strerror(r) << dendl;
+  if (r == 0) {
+    auto itc = std::find_if(
```

Comment:
Wouldn't this be present always seeing that it is pushed back on line 972 before the snap is actually created?
if the snap creation fails wouldn't m_create_group_snaps still contain an entry that actually does not exist on disk.